### PR TITLE
Lesson1

### PR DESCRIPTION
**Pull request by JuliaBal**
Completed first exercises

### CODE REVIEW

File: ConsoleApp1/Program.cs [hunk 1]
```diff
@@ -1,15 +1,118 @@
...
-using System.Threading.Tasks;
 
-namespace ConsoleApp1
+namespace Lesson1
 {
     class Program
     {
         static void Main(string[] args)
         {
+            //Exercise 1
```
Comment (NaGriS) [inline ConsoleApp1/Program.cs:9]: As described in homework issue 4 exercises should be in 4 different projects in 1 solution, but you got 1 class, should be changed

File: ConsoleApp1/Program.cs [hunk 2]
```diff
@@ -1,15 +1,118 @@
...
+            }
+
+            else if(z == 0)
+            {
+                Console.WriteLine("!!!z = 0!!! cannot be divided by 0.");
+            }
+
+            else
+            {
+                var fexpression = (int)((x + y) * (Math.Pow(z, 2) + 1));
```
Comment (NaGriS) [inline ConsoleApp1/Program.cs]: Why we have cast to int there?

File: ConsoleApp1/Program.cs [hunk 3]
```diff
@@ -1,15 +1,118 @@
...
+                Console.WriteLine("First expression: " + fexpression);
+
+                var sexpression = ((x % z) - 1) * (Math.Sqrt(y));
+                Console.WriteLine("Second expression: " + (Math.Round(sexpression, 2)));
```
Comment (NaGriS) [inline ConsoleApp1/Program.cs:35]: Same, why we are rounding it ?

File: ConsoleApp1/Program.cs [hunk 4]
```diff
@@ -1,15 +1,118 @@
...
+            Console.WriteLine("Exercise 2. Press any key");
+            Console.ReadKey();
+            Console.WriteLine("Enter length");
+            int length = Convert.ToInt32(Console.ReadLine());
+            Console.WriteLine("Enter width");
+            int width = Convert.ToInt32(Console.ReadLine());
+            Console.WriteLine("Enter height");
+            int height = Convert.ToInt32(Console.ReadLine());
+
+            if(length < 0 ^ width < 0 ^ height < 0)
```
Comment (NaGriS) [inline ConsoleApp1/Program.cs]: Why we need bit XOR there?
if you want to check all three sides there you need to use &&

File: ConsoleApp1/Program.cs [hunk 5]
```diff
@@ -1,15 +1,118 @@
...
+            {
+                Console.WriteLine("Incorrect data");
+            }
+
+            else
+            {
+                var diagonal = Math.Sqrt(Math.Pow(length, 2) + Math.Pow(width, 2) + Math.Pow(height, 2));
+                Console.WriteLine("Parallelepiped diagonal length = " + (Math.Round(diagonal, 2)));
```
Comment (NaGriS) [inline ConsoleApp1/Program.cs:60]: Why round???

File: ConsoleApp1/Program.cs [hunk 6]
```diff
@@ -1,15 +1,118 @@
...
+            Console.WriteLine("Exercise 3. Press any key");
+            Console.ReadKey();
+            Console.WriteLine("Enter string 1");
+            var str1 = Console.ReadLine().ToUpper();
+            Console.WriteLine("Enter string 2");
+            var str2 = Console.ReadLine().ToUpper();
+            Console.WriteLine("Enter string 3");
+            var str3 = Console.ReadLine().ToUpper();
+
+            if (str1.StartsWith(str2))
```
Comment (NaGriS) [inline ConsoleApp1/Program.cs:74]: Exercise 3 is solved, but the format is not equal to the described one:
should be: "String str1 starts from str2: TRUE" or "String str1 starts from str2: FALSE"

File: ConsoleApp1/Program.cs [hunk 7]
```diff
@@ -1,15 +1,118 @@
...
+
+            //Exercise 4
+            Console.WriteLine("Exercise 4. Press any key");
+            Console.ReadKey();
+            Console.WriteLine("Enter string 1");
+            String string1 = Console.ReadLine();
+            Console.WriteLine("Enter string 2");
+            String string2 = Console.ReadLine();
+            Console.WriteLine("Enter symbol");
+            char ch = Convert.ToChar(Console.ReadLine());
```
Comment (NaGriS) [inline ConsoleApp1/Program.cs:100]: What if will input more than one char?

Maybe there is a method in Console class that will help us to read the char?

File: ConsoleApp1/Program.cs [hunk 8]
```diff
@@ -1,15 +1,118 @@
...
+            String stringresult = string1 + string2;
+            int index = (stringresult.Length - 1);
+            while (index != 0)
+            {
+                if (stringresult[index] == ch)
+                {
+                    Console.WriteLine("Index of symbol in string: " + index);
```
Comment (NaGriS) [inline ConsoleApp1/Program.cs]: Format is incorrect, see exercise 4

File: ConsoleApp1/Program.cs [hunk 9]
```diff
@@ -1,15 +1,118 @@
...
+                    break;
+                }
+                else
+                    index--;
```
Comment (NaGriS) [inline ConsoleApp1/Program.cs:111]: Good example of using cycle and if, but there is only one method, described in String class that should help you do this in one line. Will be easier to read and 1 line of code < than 9

File: ConsoleApp1/Program.cs (hunk 3 above)
Comment (JuliaBal) [reply ConsoleApp1/Program.cs:35]: because there can be a number with a long fractional part.

File: ConsoleApp1/Program.cs (hunk 4 above)
Comment (JuliaBal) [reply ConsoleApp1/Program.cs]: Sorry, I mixed up the operators...

File: ConsoleApp1/Program.cs (hunk 7 above)
Comment (JuliaBal) [reply ConsoleApp1/Program.cs:100]: I tried to use the console.readkey, but every time the program stopped at this point.

**Pull request merged**